### PR TITLE
rgw_sal_motr: [CORTX- 31606] : Enabling versioning support for Object Tagging APIs

### PR DESCRIPTION
**Pull request by shraddhaghatol**
Problem: Enabled tagging APIs put-object-tagging,get-object-tagging,delete-object tagging APIs on version enable bucket.
Solution: Handled all the version/ null version cases for the tagging APIs.

Signed-off-by: shraddha ghatol <shraddha.j.ghatol@seagate.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by shraddhaghatol** [comment]
Addressed all the review comments from this PR -https://github.com/Seagate/cortx-rgw/pull/236

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -1340,53 +1361,31 @@ int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx,
...
-  } else {
-    bname = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
-    key   = this->get_key().to_str();
-  }
-  ldpp_dout(dpp, 20) << __func__ << ": for bucket " << bname << "/" << key << dendl;
-}
+  string bname, key;
+  fetch_obj_entry_and_key(dpp, ent, bname, key, target_obj);
 
-int MotrObject::read_obj_attrs(const DoutPrefixProvider* dpp, std::string& bname, std::string& key, bufferlist& bl, rgw_obj* target_obj)
```
Comment (cdeshmukh) [inline src/rgw/rgw_sal_motr.cc:1374]: same here, lets keep this function separate

File: src/rgw/rgw_sal_motr.cc [hunk 2]
```diff
@@ -1296,36 +1296,57 @@ MotrObject::~MotrObject() {
...
+     ldpp_dout(dpp, 20) <<__func__<< ": fetching null version key instance for latest object " << key << dendl;
+  }
+
+   return 0;
+}
+
 int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx, Attrs* setattrs, Attrs* delattrs, optional_yield y, rgw_obj* target_obj)
 {
-  // TODO : Set tags for multipart objects
+   // TODO : Set tags for multipart objects
```
[the remark this answers is not in the document]
Comment (shraddhaghatol) [reply src/rgw/rgw_sal_motr.cc]: Sure. Corrected it.

File: src/rgw/rgw_sal_motr.cc [hunk 3]
```diff
@@ -1340,53 +1361,31 @@ int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx,
...
   rgw_bucket_dir_entry ent;
-  bufferlist& blr = bl;
-  auto iter = blr.cbegin();
-  ent.decode(iter);
-  decode(attrs, iter);
-
-  return 0;
-}
-
-void MotrObject::read_bucket_info(const DoutPrefixProvider* dpp, std::string& bname, std::string& key, rgw_obj* target_obj)
```
[the remark this answers is not in the document]
Comment (shraddhaghatol) [reply src/rgw/rgw_sal_motr.cc]: added the function again

File: src/rgw/rgw_sal_motr.cc (hunk 1 above)
Comment (shraddhaghatol) [reply src/rgw/rgw_sal_motr.cc:1374]: we do not need this function instead we are using get_bucket_dir_ent()

Comment (cdeshmukh) [reply src/rgw/rgw_sal_motr.cc:1374]: alright, didn't know we had a function with similar functionality already

**Review by cdeshmukh — APPROVED** [review]

**Review by sachitanands — APPROVED** [review]

**Review by priyanka25081999 — APPROVED** [review]

**Review by shriya-deshmukh — APPROVED** [review]

**Review by selvakumaar5496 — APPROVED** [review]
LGTM

**Comment by shraddhaghatol** [comment]
Test cases are captured on -
https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/1036157829/WIP+Test+scenarios+for+overall+versioning+feature#Versioning-cases-with-tagging-APIs%3A%3A

**Pull request merged**